Content.Server/_DV/CosmicCult/CosmicCultSystem.cs

```diff
@@ -0,0 +1,209 @@
+        SubscribeLocalEvent<CosmicImposingComponent, ComponentInit>(OnStartImposition);
+        SubscribeLocalEvent<CosmicImposingComponent, ComponentRemove>(OnEndImposition);
+        SubscribeLocalEvent<CosmicImposingComponent, RefreshMovementSpeedModifiersEvent>(OnImpositionMoveSpeed);
+
+        SubscribeLocalEvent<CosmicCultExamineComponent, ExaminedEvent>(OnCosmicCultExamined);
+
+        SubscribeFinale(); //Hook up the cosmic cult finale system
+    }
+
+    public void MalignEcho(Entity<CosmicCultComponent> uid)
+    {
+        if (_cultRule.AssociatedGamerule(uid) is not {} cult)
+            return;
+        if (cult.Comp.CurrentTier > 1 && !_random.Prob(0.5f))
+            Spawn("CosmicEchoVfx", Transform(uid).Coordinates);
+    }
+
+    #region Housekeeping
+
+    /// <summary>
+    /// Creates the Cosmic Void pocket dimension map.
+    /// </summary>
+    // Rogue Ascendants use this too, which are generalized MidRoundAntags, so we keep the map around. If you're porting cosmic cult, and do not want rogue ascendants, feel free to move this into selective usage akin to NukeOps base.
+    private void OnRoundStart(RoundStartingEvent ev)
+    {
+        if (_mapLoader.TryLoadMap(_mapPath, out var map, out _, new DeserializationOptions { InitializeMaps = true }))
+            _map.SetPaused(map.Value.Comp.MapId, false);
+    }
+
+    private void OnCosmicCultExamined(Entity<CosmicCultExamineComponent> ent, ref ExaminedEvent args)
+    {
+        if (HasComp<CosmicCultComponent>(args.Examiner))
+            args.PushMarkup(Loc.GetString(ent.Comp.CultistText));
+        else
+            args.PushMarkup(Loc.GetString(ent.Comp.OthersText));
+    }
+    #endregion
+
+    #region Init Cult
+    /// <summary>
+    /// Add the starting powers to the cultist.
+    /// </summary>
+    private void OnStartCultist(Entity<CosmicCultComponent> uid, ref ComponentInit args)
+    {
+        foreach (var actionId in uid.Comp.CosmicCultActions)
+        {
+            var actionEnt = _actions.AddAction(uid, actionId);
+            uid.Comp.ActionEntities.Add(actionEnt);
+        }
+        _eye.RefreshVisibilityMask(uid.Owner);
+        _alerts.ShowAlert(uid, uid.Comp.EntropyAlert);
+    }
+
+    /// <summary>
+    /// Add the Monument summon action to the cult lead.
+    /// </summary>
+    private void OnStartCultLead(Entity<CosmicCultLeadComponent> uid, ref ComponentInit args)
+    {
+        _actions.AddAction(uid, ref uid.Comp.CosmicMonumentPlaceActionEntity, uid.Comp.CosmicMonumentPlaceAction, uid);
+    }
+
+    private void OnGetVisMask(Entity<CosmicCultComponent> uid, ref GetVisMaskEvent args)
+    {
+        args.VisibilityMask |= (int)VisibilityFlags.CosmicCultMonument;
+    }
+
+    /// <summary>
+    /// Called by Cosmic Siphon. Increments the Cult's global objective tracker.
+    /// </summary>
+    #endregion
+
+
+    #region Equipment Pickup
```

Comment:
Remove extra blank line
```suggestion
    #endregion

    #region Equipment Pickup
```